tests/openssl/CMakeLists.txt

```
@@ -122,576 +189,713 @@ foreach (testcase ${buildtests})
   endif ()
 endforeach (testcase)
 
+# Add buildtests for OpenSSL_3
+file(STRINGS "tests_3.supported.buildtest" buildtests_3)
+foreach (testcase ${buildtests_3})
+  get_testcase_name(${testcase} name "../../3rdparty/openssl/openssl_3/test/")
+  add_enclave_test(tests/openssl_3/${name} openssl_host openssl-${name}_3_enc
+                   ${name})
+  # OpenSSL_3 does not support symcrypt yet.
+endforeach (testcase)
+
 file(STRINGS "tests.unsupported.symcrypt" UNSUPPORTED_SYMCRYPT_TESTS)
 
 # Helper for adding tests with arguments.
-macro (add_openssl_test TEST_NAME NAME)
-  add_enclave_test(tests/openssl/${TEST_NAME} openssl_host openssl-${NAME}_enc
-                   ${TEST_NAME} ${ARGN})
-  if ((NOT "${TEST_NAME}" IN_LIST UNSUPPORTED_SYMCRYPT_TESTS)
-      AND (ENABLE_SYMCRYPT_OPENSSL_TESTS))
-    add_enclave_test(tests/openssl/${TEST_NAME}_symcrypt openssl_host
-                     openssl-${NAME}_symcrypt_enc ${TEST_NAME} ${ARGN})
+macro (add_openssl_test OPENSSL3 TEST_NAME NAME)
+  if (OPENSSL3)
+    add_enclave_test(tests/openssl_3/${TEST_NAME} openssl_host
+                     openssl-${NAME}_3_enc ${TEST_NAME} ${ARGN})
```

```suggestion
                     openssl-3-${NAME}_enc ${TEST_NAME} ${ARGN})
```
Also see the comment on enc/CMakeLists.txt